eclair-core/src/test/scala/fr/acinq/eclair/io/PeerSpec.scala

```diff
@@ -137,6 +136,31 @@ class PeerSpec extends TestkitBaseClass with StateTestsHelperMethods {
     probe.expectMsg("disconnecting")
   }
 
+  test("handle new connection in state CONNECTED") { f =>
```

Comment:
Nice test!

Reply:
I think I can actually improve it a little bit.

Reply:
Done with b25a040. Testing only via the deatch watch was pretty indirect.

Reply:
Actually that's the reason why I had to do two rounds (because I needed to check that `peerConnection2` was killed). I could get rid of `peerConnection3` now. Well, let's leave it that way.

Reply:
I agree, better to keep `peerConnection3` it makes it even more explicit.